tests/framework/extensions/machinepools/harvester_machine_config.go

```diff
@@ -0,0 +1,32 @@
+package machinepools
+
+import "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+const (
+	HarvesterKind           = "HarvesterConfig"
+	HarvesterPoolType       = "rke-machine-config.cattle.io.harvesterconfig"
+	HarvesterResourceConfig = "harvesterconfigs"
+)
+
+// NewHarvesterMachineConfig is a constructor to set up rke-machine-config.cattle.io.harvesterconfig.
+// It returns an *unstructured.Unstructured that CreateMachineConfig uses to created the rke-machine-config
+func NewHarvesterMachineConfig(generatedPoolName, namespace, network, image string) *unstructured.Unstructured {
+	machineConfig := &unstructured.Unstructured{}
+	machineConfig.SetAPIVersion("rke-machine-config.cattle.io/v1")
+	machineConfig.SetKind(HarvesterKind)
+	machineConfig.SetGenerateName(generatedPoolName)
+	machineConfig.SetNamespace("fleet-default")
+
+	machineConfig.Object["diskSize"] = "40"
+	machineConfig.Object["cpuCount"] = "2"
```

Comment:
I think we'll probably have these in a config file. But my PR will have those details and you build off of that work